.ci/test_r_package.sh

```diff
@@ -230,18 +219,12 @@ if [[ $R_BUILD_TYPE == "cmake" ]]; then
         cat $BUILD_LOG_FILE \
         | grep --count "R version passed into FindLibR.cmake: ${R_VERSION}"
     )
-    if [[ $used_correct_r_version -ne 1 ]]; then
+    if [[ $passed_correct_r_version_to_cmake -ne 1 ]]; then
```

Comment:
This was a fix found in #6501. That PR was abandoned accidentally and I'd still like to give @Kunal-Singh-Dadhwal an opportunity to submit that work again.

But I want this particular fix, to get more confidence that the R CI jobs are working as expected.

context: https://github.com/microsoft/LightGBM/pull/6501#discussion_r1649202588

Reply:
@jameslamb Thanks for the opportunity will make a commit really soon